documentation/src/main/asciidoc/userguide/chapters/query/hql/QueryLanguage.adoc

```diff
@@ -1022,13 +1025,33 @@ Next, functions for working with numeric values:
 | `exp()` | Exponential function | `exp(x)` | Universal in SQL dialects
 | `power()` | Exponentiation | `power(x,y)` | Universal in SQL dialects
 | `ln()` | Natural logarithm | `ln(x)` | Very common in SQL dialects
+| `log()` | Base-N logarithm | `log(x,x)` | Very common in SQL dialects
+| `log2()` | Base-2 logarithm | `log2(x)` | Very common in SQL dialects
 | `log10()` | Base-10 logarithm | `log10(x)` | Very common in SQL dialects
-| `sin()`, `cos()`, `tan()`, `asin()`, `acos()`, `atan()`, `atan2()` | Basic trigonometric functions | `sin(theta)`, `cos(theta)`, `atan2(opposite, adjacent)` | Very common in SQL dialects
+| `sin()`, `cos()`, `tan()`, `cot()`, `asin()`, `acos()`, `atan()`, `atan2()`, `acot()` | Basic trigonometric functions | `sin(theta)`, `cos(theta)`, `atan2(opposite, adjacent)` | Very common in SQL dialects
+| `sinh()`, `cosh()`, `tanh()`, `coth()`, `asinh()`, `acosh()`, `atanh()`, `acoth()` | Basic trigonometric functions | `sinh(theta)`, `cosh(theta)` | Very common in SQL dialects
```

Comment:
Rollback please.

Reply:
1. Hyperbolic functions aren't supported on all of the built-in SQL dialects, so they don't belong on this list.
2. `acot()` doesn't even exist.
3. Looks like you could add `cot()` if you feel you really need it.

Reply:
FTR, the hyperbolic functions can be emulated, of course, using `exp()`. But someone would need to do that work.